lib/src/facts/posix/uptime_resolver.cc

```diff
@@ -43,6 +48,27 @@ namespace facter { namespace facts { namespace posix {
 
     int64_t uptime_resolver::get_uptime()
     {
+        // We first try to get the system boot time by reading it directly from
+        // the utmp file using the getutxtent system call. This approach
+        // was adapted from https://stackoverflow.com/questions/2071689/how-can-i-get-the-uptime-of-a-ibm-aix-box-in-seconds?answertab=votes#tab-top
+        LOG_DEBUG("Attempting to calculate the uptime from the utmp file...");
+        setutxent();
```

Comment:
The whole `setutxent` and `endutxent` sets the utmp file pointer to the beginning and to the end, respectively. It is an idiomatic way of using the `getutxent` (and other utmp system calls) based on what I'm seeing in https://linux.die.net/man/3/getutxline.